### ISSUE TITLE
How to: installing the old app so that it's exempt from updates

### ISSUE THREAD

**Issue by gingerbeardman** [qnblackcat/How-to-Downgrade-apps-on-AppStore-with-iTunes-and-Charles-Proxy#26]
Thanks for the guide!

I link it from [my recent blog post](https://blog.gingerbeardman.com/2023/08/17/going-back-to-the-old-pre-x-twitter-ios-app/) where I provide one new innovation: prevent the old app from updating.

I figured out this trick back in 2016, and I'm glad it still works.

1. extract the .ipa
2. locate the Payload app
3. open your iPhone in Apple Configurator
4. drag the Payload onto your iPhone
5. done!

**Comment by qnblackcat** [comment]
Feel free to make a Pull request

**Comment by GoddeerisEdouard** [comment]
This method (of blocking updates) doesn’t seem to work for IOS 17 devices… as they “aren’t recognized” on that version of iTunes. Would love to also be able to block app updates.

**Comment by gingerbeardman** [comment]
@GoddeerisEdouard in what way does it not work? I ask because you mention iTunes, which is not involved in any way in the installation method. So, I think it will install correctly when you use Apple Configurator.

**Comment by GoddeerisEdouard** [comment]
@gingerbeardman in your old blog where you first mentioned [your way](https://blog.gingerbeardman.com/2016/07/19/how-to-prevent-an-individual-ios-app-from-updating-forever/) of blocking updates, you’re using iTunes. I haven’t tried  the updated method (above ) mentioned in this Issue where you’re using macOS yet.

**Comment by gingerbeardman** [comment]
@GoddeerisEdouard right, that old way no longer works. I'll make a note on that old blog post.

the modern way is to use iTunes to download the old version of the app, then use Apple Configurator to install. As mentioned at the top of this thread and in this newer blog post:

https://blog.gingerbeardman.com/2023/08/17/going-back-to-the-old-pre-x-twitter-ios-app/

**Comment by qnblackcat** [comment]
> @GoddeerisEdouard right, that old iTunes-only way no longer works. I'll make a note on that old blog post.
> 
> The modern way is to use iTunes only to download the old version of the app, then use Apple Configurator to install it. As mentioned at the top of this thread and in this newer blog post:
> 
> https://blog.gingerbeardman.com/2023/08/17/going-back-to-the-old-pre-x-twitter-ios-app/

This is awesome. Definitely will mention it in the guide

**Comment by GoddeerisEdouard** [comment]
Alright, so amazing news:
It did work on MacOS by following the guide above (just dragging the app folder from Payload in Apple configurator ). I had one small issue: **my MacOS wasn't up-to-date** (Apple Configurator 2.17 (latest found for me) required MacOS 14.0 to be installed... I was on MacOS 13.x, so this was my workaround:

I found [Apple Configurator 2.16](https://archive.org/details/apple-configurator) and downloaded that one on my Mac and it worked flawlessly from there. (I couldn't drag the app in Apple Configurator at first, but I just clicked the "Add" button in Apple Configurator and just went through my Mac's directory that way.

**Comment by gingerbeardman** [comment]
Good point regarding finding the correct old version of Apple Configurator for your macOS. Long ago I installed it through the Mac App Store which guarantees I'm on the most recent version (I'm still on Monterey, so it's an old one)

**Comment by Wikinger8** [comment]
Is there a way to get Apple Configurator for Windows 10?

**Comment by gingerbeardman** [comment]
No, there is no such thing as the Apple Configurator for Windows.

**Comment by qnblackcat** [comment]
@gingerbeardman 
It seems like the link to my guide in [your blog](https://blog.gingerbeardman.com/2023/08/17/going-back-to-the-old-pre-x-twitter-ios-app/) is incorrect. Could you fix it?

**Comment by GoddeerisEdouard** [comment]
> @gingerbeardman It seems like the link to my guide in [your blog](https://blog.gingerbeardman.com/2023/08/17/going-back-to-the-old-pre-x-twitter-ios-app/) is incorrect. Could you fix it?

If you're talking about
> ...
> [Follow this great step-by-step tutorial](https://github.com/qnblackcat/How-to-Downgrade-apps-on-AppStore-with-iTunes-and-Charles-Proxy). It might seem ...

The link is correct...?

**Comment by gingerbeardman** [comment]
Only because I fixed it a couple hours ago—thanks for the report it was a bug in my blogging system! @qnblackcat

**Comment by gingerbeardman** [comment]
> Just curious, what if I download from App Store after installing this way? Will my app be updated to latest version? Will my app data get lost?

1. Yes, the app will be upgraded in-place.
2. No, the data will be safe.

Doing so will just install over the existing one with the additional receipt. App data is stored in a different place so it is not touched.

**Comment by ZeppLu** [comment]
Feedback: I successfully downgraded  WeChat the other day, and observed that App Store showed "download" button for it. Then I kept auto update in App Store settings open, expecting that WeChat will stay on the old version.

But today I notice something interesting:
1. WeChat once again runs the latest version
2. App Store now shows "open" button
3. there is no WeChat in the recently updated list of App Store

**Comment by gingerbeardman** [comment]
@ZeppLu did you install the app from Payload folder (no receipt)? It seems like you installed the whole IPA including receipt. The point of all this is to not install the receipt so the App Store app has no knowledge of the app.

**Comment by ZeppLu** [comment]
@gingerbeardman I'm pretty sure I've extracted tha .ipa file and added the .app file in Apple Configurator. And I've confirmed that App Store thought WeChat was not yet downloaded.

My hypothesis is that App Store won't show red dots for the downgraded app if auto update is turned off, but it will still update it with auto update on? Anyway, thanks for your help, but I don't want to continue messing with it.

**Comment by gingerbeardman** [comment]
No, that's an incorrect hypothesis. The only two scenarios are:
1. receipt
  - app store knows about it
  - will auto update
2. no receipt
  - app store has no knowledge of it
  - never updates 

Take care!

**Comment by gingerbeardman** [comment]
> UPDATE: I retried, and this time successfully keep WeChat in old version

Good news!

> I think the point here is that you **MUST NOT** enable App Store auto update after old version is installed. That is, if you want to enable auto update, do it **BEFORE** following all these steps. Hope this tip helps future viewers.

No, this is another incorrect assertion. If the app is installed without receipt, then it's impossible for the App Store to know that it is there.

**Comment by alneven** [comment]
> so this was my workaround: I found [Apple Configurator 2.16](https://archive.org/details/apple-configurator) and downloaded that one on my Mac and it worked flawlessly from there. (I couldn't drag the app in Apple Configurator at first, but I just clicked the "Add" button in Apple Configurator and just went through my Mac's directory that way.


For me, neither dragging and dropping into Apple Configurator 2.18 nor using the '+' button to select the payload content is working.

![Image](https://github.com/user-attachments/assets/fe904a3d-619c-451f-8d73-0de6a10ce8d1)

I got the IPA file, extracted it after renamed to ZIP
and I have the Payload folder

my Ipad is under iPadOS 18.3.2

**Comment by gingerbeardman** [comment]
You need to select the file/content inside of the Payload folder, seen to the top right of your screenshot. @alneven

**Comment by alneven** [comment]
> You need to select the file/app/content inside of the Payload folder, seen in top right of your screenshot. [@alneven](https://github.com/alneven)

That part is greyed out. I cannot click on the file itself

NameoftheApp.app

**Comment by alneven** [comment]
<img width="237" alt="Image" src="https://github.com/user-attachments/assets/40759c82-de62-41e0-a56a-e0a4f740227e" />

OK, the Apple Configurator version was the issue
with 2.16 it's working, with the 2.18 it isn't

**Comment by binnichtaktiv** [comment]
> Yes, this issue is mentioned earlier in the thread.

The link to apple configurator 2.16 is dead :(

**Comment by gingerbeardman** [comment]
> > Yes, this issue is mentioned earlier in the thread.
> 
> The link to apple configurator 2.16 is dead :(

Google returned a reddit thread with: https://www.mediafire.com/file/pi0rkebpah5he88/Apple_Configurator2.16.zip/file

**Comment by binnichtaktiv** [comment]
> > > Yes, this issue is mentioned earlier in the thread.
> > 
> > 
> > The link to apple configurator 2.16 is dead :(
> 
> Google returned a reddit thread with: https://www.mediafire.com/file/pi0rkebpah5he88/Apple_Configurator2.16.zip/file

Thank you I used duckduckgo and I didnt find anything useful
